pkg/core/blockchain.go

```diff
@@ -566,13 +566,19 @@ func (bc *Blockchain) jumpToStateInternal(p uint32, stage stateJumpStage) error
 		// After current state is updated, we need to remove outdated state-related data if so.
 		// The only outdated data we might have is genesis-related data, so check it.
 		if p-bc.config.MaxTraceableBlocks > 0 {
-			cache := bc.dao.GetWrapped()
+			cache := bc.dao.GetWrapped().(*dao.Simple)
 			writeBuf.Reset()
 			err := cache.DeleteBlock(bc.headerHashes[0], writeBuf)
 			if err != nil {
 				return fmt.Errorf("failed to remove outdated state data for the genesis block: %w", err)
 			}
-			// TODO: remove NEP-17 transfers and NEP-17 transfer info for genesis block, #2096 related.
+			prefixes := []byte{byte(storage.STNEP11Transfers), byte(storage.STNEP17Transfers), byte(storage.STTokenTransferInfo)}
+			for i := range prefixes {
+				cache.Store.Seek(storage.SeekRange{Prefix: prefixes[i : i+1]}, func(k, v []byte) bool {
+					_ = cache.Store.Delete(k) // It's MemCachedStore which never returns an error.
+					return true
+				})
+			}
```

Comment:
Not sure is it connected, but there was a test failing (likely due to race):
```
2022-02-16T08:54:35.9656069Z === RUN   TestStateSyncModule_RestoreBasicChain
2022-02-16T08:54:35.9656952Z     logger.go:130: 2022-02-16T08:54:34.348Z	INFO	initial gas supply is not set or wrong, setting default value	{"InitialGASSupply": "52000000"}
2022-02-16T08:54:35.9658054Z     logger.go:130: 2022-02-16T08:54:34.348Z	INFO	P2PNotaryRequestPayloadPool size is not set or wrong, setting default value	{"P2PNotaryRequestPayloadPoolSize": 1000}
2022-02-16T08:54:35.9659053Z     logger.go:130: 2022-02-16T08:54:34.348Z	INFO	MaxBlockSize is not set or wrong, setting default value	{"MaxBlockSize": 262144}
2022-02-16T08:54:35.9660011Z     logger.go:130: 2022-02-16T08:54:34.348Z	INFO	MaxBlockSystemFee is not set or wrong, setting default value	{"MaxBlockSystemFee": 900000000000}
2022-02-16T08:54:35.9661096Z     logger.go:130: 2022-02-16T08:54:34.348Z	INFO	MaxTransactionsPerBlock is not set or wrong, using default value	{"MaxTransactionsPerBlock": 512}
2022-02-16T08:54:35.9662158Z     logger.go:130: 2022-02-16T08:54:34.348Z	INFO	MaxValidUntilBlockIncrement is not set or wrong, using default value	{"MaxValidUntilBlockIncrement": 5760}
2022-02-16T08:54:35.9663006Z     logger.go:130: 2022-02-16T08:54:34.349Z	INFO	no storage version found! creating genesis block
2022-02-16T08:54:35.9663576Z     helper_test.go:274: native GAS hash: cf76e28bd0062c4a478ee35561011319f3cfa4d2
2022-02-16T08:54:35.9664140Z     helper_test.go:275: native NEO hash: f563ea40bc283d4d0e05c48ea305b3f2a07340ef
2022-02-16T08:54:35.9732917Z     helper_test.go:276: native Policy hash: 7bc681c0a1f71d543457b68bba8d5f9fdd4e5ecc
2022-02-16T08:54:35.9733622Z     helper_test.go:277: native Notary hash: 3bec3531119bbad76dd044920b0de6c3194fe1c1
2022-02-16T08:54:35.9734368Z     helper_test.go:278: Block0 hash: 4b21c3ac87ae73f2862905604123d341ae5f4b3da8fd4a1f26c5bc3ade7322c1
2022-02-16T08:54:35.9734801Z panic: assignment to entry in nil map
2022-02-16T08:54:35.9735002Z 
2022-02-16T08:54:35.9735122Z goroutine 6107 [running]:
2022-02-16T08:54:35.9735767Z github.com/nspcc-dev/neo-go/pkg/core/storage.(*MemoryStore).put(...)
2022-02-16T08:54:35.9736358Z 	/home/runner/work/neo-go/neo-go/pkg/core/storage/memory_store.go:54
2022-02-16T08:54:35.9737002Z github.com/nspcc-dev/neo-go/pkg/core/storage.(*MemoryStore).PutChangeSet(0xc000612c80, 0xc000028ec7)
2022-02-16T08:54:35.9737626Z 	/home/runner/work/neo-go/neo-go/pkg/core/storage/memory_store.go:92 +0x129
2022-02-16T08:54:35.9738198Z github.com/nspcc-dev/neo-go/pkg/core/storage.(*MemoryStore).PutBatch(...)
2022-02-16T08:54:35.9738785Z 	/home/runner/work/neo-go/neo-go/pkg/core/storage/memory_store.go:85
2022-02-16T08:54:35.9739399Z github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).removeOldStorageItems(0xc0000c3400)
2022-02-16T08:54:35.9739995Z 	/home/runner/work/neo-go/neo-go/pkg/core/blockchain.go:502 +0x309
2022-02-16T08:54:35.9740580Z created by github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).jumpToStateInternal
2022-02-16T08:54:35.9741474Z 	/home/runner/work/neo-go/neo-go/pkg/core/blockchain.go:562 +0x6f4
2022-02-16T08:54:35.9741960Z FAIL	github.com/nspcc-dev/neo-go/pkg/core	19.401s
```

Reply:
See https://github.com/nspcc-dev/neo-go/runs/5213412866?check_suite_focus=true.

Reply:
That can only happen after `(*MemoryStore).Close`, so it's certainly not related, but it reminds me of #2325.

Reply:
It happens because `removeOldStorageItems()` is asynchronous and it's not holding any of the locks we have. So `jumpToState()` finishes successfully making it possible to `Close()`, but `removeOldStorageItems()` is not finished yet.

Reply:
OK, let's fix it in a separate issue.

Reply:
Let's just fix it.